action/protocol/execution/evm/evm.go

```diff
@@ -143,12 +143,23 @@ func ExecuteContract(
 ) ([]byte, *action.Receipt, error) {
 	raCtx := protocol.MustGetRunActionsCtx(ctx)
 	hu := config.NewHeightUpgrade(&raCtx.Genesis)
-	stateDB := NewStateDBAdapter(
-		sm,
-		raCtx.BlockHeight,
-		hu.IsPre(config.Aleutian, raCtx.BlockHeight),
-		execution.Hash(),
-	)
+	var stateDB *StateDBAdapter
+	if raCtx.History {
+		stateDB = NewStateDBAdapter(
+			sm,
```

Comment:
we passed in the sm, which was a working set. Somehow, we read the "history" value from the working set and assign it to raCtx, then use it here eventually.